src/Apps/CollectorProfile/Routes/Saves2/Components/Actions/EditSavesModal.tsx

```diff
@@ -0,0 +1,177 @@
+import {
+  Button,
+  Flex,
+  LabeledInput,
+  ModalDialog,
+  Spacer,
+  Text,
+  useToasts,
+} from "@artsy/palette"
+import EditIcon from "@artsy/icons/EditIcon"
+import { useTranslation } from "react-i18next"
+import { SavesArtworks_collection$data } from "__generated__/SavesArtworks_collection.graphql"
+import { Form, Formik, FormikHelpers } from "formik"
+import * as Yup from "yup"
+import { pick } from "lodash"
+import { useMutation } from "Utils/Hooks/useMutation"
+import { graphql } from "react-relay"
+import { EditSavesModalMutation } from "__generated__/EditSavesModalMutation.graphql"
+import createLogger from "Utils/logger"
+
+interface Props {
+  collection: SavesArtworks_collection$data
+  setIsEditModalOpen: React.Dispatch<React.SetStateAction<boolean>>
+}
+
+interface FormikValues {
+  name: string
+}
+
+const MAX_NAME_LENGTH = 40
+
+const logger = createLogger("EditSavesModal")
+
+export const EditSavesModal: React.FC<Props> = ({
+  collection,
+  setIsEditModalOpen,
+}) => {
+  const { t } = useTranslation()
+
+  const initialValues = pick(collection, "name")
+
+  const validationSchema = Yup.object().shape({
+    name: Yup.string()
+      .required(t("collectorSaves.editListModal.fields.name.required")) // TODO: confirm copy
+      .max(MAX_NAME_LENGTH),
+  })
+
+  const { submitMutation } = useMutation<EditSavesModalMutation>({
```

Comment:
The pattern we've started adopting around all of our relay codebases follows this pretty closely: https://github.com/artsy/force/blob/main/src/Apps/MyCollection/Routes/EditArtwork/Mutations/useCreateArtwork.ts#L5; we define it in a hook in its own file.